File: src/main/java/io/bindernews/thegrackle/power/TargetingComputerPower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.AbstractPower;
import com.megacrit.cardcrawl.powers.watcher.VigorPower;
import io.bindernews.thegrackle.GrackleMod;

public class TargetingComputerPower extends BasePower {
    public static final String POWER_ID = GrackleMod.makeId(TargetingComputerPower.class);

    public TargetingComputerPower(AbstractCreature owner, int amount) {
        super(POWER_ID);
        setOwnerAmount(owner, amount);
        loadRegion("lockon");
    }

    @Override
    public void atStartOfTurn() {
        addToBot(new ApplyPowerAction(owner, owner, new VigorPower(owner, amount)));
    }

    @Override
    public void updateDescription() {
        description = formatDesc(0, amount);
    }

    @Override
    public AbstractPower makeCopy() {
        return new TargetingComputerPower(owner, amount);
    }
}

File: src/main/kotlin/net/bindernews/grackle/power/TargetingComputerPower.kt
package net.bindernews.grackle.power

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.powers.watcher.VigorPower
import net.bindernews.grackle.GrackleMod

class TargetingComputerPower(owner: AbstractCreature, amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, amount)
        loadRegion("lockon")
    }

    override fun atStartOfTurn() {
        addToBot(ApplyPowerAction(owner, owner, VigorPower(owner, amount)))
    }

    override fun updateDescription() {
        description = formatDesc(0, amount)
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(TargetingComputerPower::class.java)
    }
}